client/rest-high-level/build.gradle

```diff
@@ -26,9 +26,9 @@ group = 'org.elasticsearch.client'
 dependencies {
   compile "org.elasticsearch:elasticsearch:${version}"
   compile "org.elasticsearch.client:rest:${version}"
-  compile "org.elasticsearch.plugin:parent-join-client:${version}"
-  compile "org.elasticsearch.plugin:aggs-matrix-stats-client:${version}"
```

Comment:
Yes, I think we could avoid to ship them by default and let users add the JAR files to the classpath if they need them.